sforzando/q-lako
# Register item in Airtable

## Describe the PR

To close #7.
Register item in Airtable.

## Screenshots

<img width="1904" alt="Screen Shot 2020-11-22 at 14 53 32" src="https://user-images.githubusercontent.com/67737076/99896290-89b49680-2cd2-11eb-8098-0f6a99c82372.png">


## Detail of the change

Implement the ability to register a dictionary with the appropriate key and value to Airtable.

### Asset dataclass

|Instance name |  type | Example |
| ---- | ---- | ---- |
| title | str | PlayStation 5 (CFI-1000A01) | 
| asset_id | int | 1606051826 | 
| asin | str | B08GGGBKRQ |
| images | list | [{ "url": "https://images-na.ssl-images-amazon.com/images/I/61YYOeZy9aL._AC_SL1500_.jpg"}] |
| url | str | https://www.amazon.co.jp/ソニー・インタラクティブエンタテインメント-PlayStation-5-CFI-1000A01/dp/B08GGGBKRQ/ |
| manufacture | str | ソニー・インタラクティブエンタテインメント |
| contributor | str | None |
| product_group | str | Video game |
| publication_date | ISO 8601 formatted str | 2020-11-18T23:00+09:00 |
| features | str | 圧巻のスピード:統合I/O(Integrated I/O)により、カスタムされたCPU・GPU・SSDがその力を発揮。 |
| default_position | str | sforzando 川崎 |
| current_position | str | 渡邉宅 |
| note | str | リモートゲーム大会用に購入。 |
| registrant_name | str | yusuke-sforzando |
| registered_at | ISO 8601 formatted str | 2020-11-20 02:01:01.847113+09:00 |


Also see examples of key values and values of registrable dictionaries.

```test_airtable_client.py

registerable_asset = Asset(title="PlayStation 5 (CFI-1000A01)",
                               asin="B08GGGBKRQ",
                               url="https://www.amazon.co.jp/ソニー・インタラクティブエンタテインメント-PlayStation-5-CFI-1000A01/dp/B08GGGBKRQ/",
                               images=[{"url": "https://images-na.ssl-images-amazon.com/images/I/61YYOeZy9aL.AC_SL1500.jpg"}],
                               manufacture="ソニー・インタラクティブエンタテインメント",
                               contributor="None",
                               product_group="Video game",
                               publication_date="2020-11-14 02:01:01.847113+09:00",
                               features="圧巻のスピード:統合I/O(Integrated I/O)により、カスタムされたCPU・GPU・SSDがその力を発揮。",
                               default_position="sforzando 川崎",
                               current_position="渡邉宅",
                               note="リモートゲーム大会用に購入。",
                               registrant_name="yusuke-sforzando",
                               registered_at="2020-11-20 02:01:01.847113+09:00")


```

## Anticipated impacts

After making the registration to Airtable, you do not remove it, so you need to remove it manually.

## To reproduce

Steps to check the behavior:

1. Check the pre-registration status of [Airtable](https://airtable.com/tbl3fqXr9nJmLm8CM/viwc0kRm6kEoxk5EW?blocks=hide#)

1. `pip install --upgrade -r requirements.txt`

1. Run the test code and go through the registration process

    `pytest -vv .`

1. Check `Airtable` and make sure it is registered correctly
1. This PR does not support up to the deletion of registered items, so you have to manually remove the registered values in  `Airtable`

## Additional context

This PR additionally requires the installation of the following Python libraries.

- [datetime](https://docs.python.org/ja/3/library/datetime.html#datetime.time.isoformat)
- [airtable-python-wrapper](https://github.com/gtalarico/airtable-python-wrapper)
